lib/handlers/quote-to-ratio/quote-to-ratio.ts

```diff
@@ -0,0 +1,422 @@
+        maxIterations,
+      },
+      requestInjected: {
+        router,
+        log,
+        id: quoteId,
+        chainId,
+        tokenProvider,
+        tokenListProvider,
+        poolProvider,
+        metric,
+      },
+    } = params;
+
+    // Parse user provided token address/symbol to Currency object.
+    const before = Date.now();
+
+    const type = 'exactIn';
+
+    const token0 = await tokenStringToCurrency(
+      tokenListProvider,
+      tokenProvider,
+      token0Address,
+      token0ChainId,
+      log
+    );
+
+    const token1 = await tokenStringToCurrency(
+      tokenListProvider,
+      tokenProvider,
+      token1Address,
+      token1ChainId,
+      log
+    );
+
+    metric.putMetric(
+      'Token01StrToToken',
+      Date.now() - before,
+      MetricLoggerUnit.Milliseconds
+    );
+
+    if (!token0) {
+      return {
+        statusCode: 400,
+        errorCode: 'TOKEN0_INVALID',
+        detail: `Could not find token with address "${token0Address}"`,
+      };
+    }
+
+    if (!token1) {
+      return {
+        statusCode: 400,
+        errorCode: 'TOKEN1_INVALID',
+        detail: `Could not find token with address "${token1Address}"`,
+      };
+    }
+
+    if (token0ChainId != token1ChainId) {
+      return {
+        statusCode: 400,
+        errorCode: 'TOKEN_CHAINS_DIFFERENT',
+        detail: `Cannot request quotes for tokens on different chains`,
+      };
+    }
+
+    if (token0.equals(token1)) {
+      return {
+        statusCode: 400,
+        errorCode: 'TOKEN_IN_OUT_SAME',
+        detail: `tokenIn and tokenOut must be different`,
+      };
+    }
+
+    const routingConfig = {
+      ...DEFAULT_ROUTING_CONFIG,
+      ...(minSplits ? { minSplits } : {}),
+    };
+
+    let swapParams: SwapConfig | undefined = undefined;
+
+    if (slippageTolerance && deadline && recipient) {
+      const slippagePer10k = Math.round(parseFloat(slippageTolerance) * 100);
+      const slippageTolerancePercent = new Percent(slippagePer10k, 10_000);
+      swapParams = {
+        deadline: Math.floor(Date.now() / 1000) + parseInt(deadline),
+        recipient: recipient,
+        slippageTolerance: slippageTolerancePercent,
+      };
+    }
+
+    const token0Balance = CurrencyAmount.fromRawAmount(
+      token0,
+      JSBI.BigInt(token0BalanceRaw)
+    );
+    const token1Balance = CurrencyAmount.fromRawAmount(
+      token1,
+      JSBI.BigInt(token1BalanceRaw)
+    );
+    const poolAccessor = await poolProvider.getPools([[
+      token0.wrapped,
+      token1.wrapped,
+      feeAmount,
+    ]])
+    const pool = poolAccessor.getPool(
+      token0.wrapped,
+      token1.wrapped,
+      feeAmount
+    ) as unknown as Pool
+    if (!pool) {
+      log.error(
+        `Could not find pool.`
+      );
+      return { statusCode: 400, errorCode: 'Could not find pool.' };
+    }
+    const position = new Position({
+      pool,
+      tickLower,
+      tickUpper,
+      liquidity: 1,
```

Comment:
hmm...it's weird because liquidity plays absolutely no role in any calculation. we define a position to get the bounds, in many cases this will be a hypothetical position with no liquidity yet.